packages/Checkbox/src/Checkbox.js

```diff
@@ -0,0 +1,81 @@
+import React from "react";
+import PropTypes from "prop-types";
+import uuid from "uuid/v4";
+import { ShirtSizes } from "@paprika/helpers/lib/customPropTypes";
+import CheckIcon from "@paprika/icon/lib/Check";
+import DashIcon from "@paprika/icon/lib/Dash";
+import checkboxStyles from "./Checkbox.styles";
+
+export const checkboxStates = {
+  CHECKED: "checked",
+  UNCHECKED: "unchecked",
+  INDETERMINATE: "indeterminate",
+};
+
+const propTypes = {
+  a11yText: PropTypes.string,
+  checkedState: PropTypes.oneOf(Object.values(checkboxStates)),
+  children: PropTypes.node,
+  isDisabled: PropTypes.bool,
+  onChange: PropTypes.func.isRequired,
+  size: PropTypes.oneOf(ShirtSizes.DEFAULT),
+};
+
+const defaultProps = {
+  a11yText: null,
+  checkedState: checkboxStates.UNCHECKED,
+  children: null,
+  isDisabled: false,
+  size: ShirtSizes.MEDIUM,
+};
+
+const Checkbox = props => {
+  const { a11yText, children, isDisabled, checkedState, size, ...moreProps } = props;
+  const { CHECKED, INDETERMINATE } = checkboxStates;
+
+  const checkboxId = React.useRef(uuid()).current;
+  const inputRef = React.useRef(null);
+
+  React.useEffect(() => {
+    if (!inputRef.current) return;
+
+    if (checkedState === INDETERMINATE) {
+      inputRef.current.indeterminate = INDETERMINATE;
+    } else {
+      inputRef.current.indeterminate = false;
+      inputRef.current.checked = checkedState === CHECKED;
```

Comment:
@mikrotron Why do you think setting the `checked` value here is necessary? Won't it handled through https://github.com/acl-services/paprika/pull/199/files#diff-70cab65379b60a6be2550b5a72d611dfR61. I think only the `indeterminate` is required to be set manually